galleries/examples/statistics/histogram_normalization.py

```diff
@@ -0,0 +1,253 @@
+  - and assign weights to the data points, so that each data point affects the
+    count in its bin differently.
+
+The Matplotlib ``hist`` method calls `numpy.histogram` and plots the results,
+therefore users should consult the numpy documentation for a definitive guide.
+
+Histograms are created by defining bin edges, and taking a dataset of values
+and sorting them into the bins, and counting or summing how much data is in
+each bin.  In this simple example, 9 numbers between 1 and 4 are sorted into 3
+bins:
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+rng = np.random.default_rng(19680801)
+
+xdata = np.array([1.2, 2.3, 3.3, 3.1, 1.7, 3.4, 2.1, 1.25, 1.3])
+xbins = np.array([1, 2, 3, 4])
+
+# changing the style of the histogram bars just to make it
+# very clear where the boundaries of the bins are:
+style = {'facecolor': 'none', 'edgecolor': 'C0', 'linewidth': 3}
+
+fig, ax = plt.subplots()
+ax.hist(xdata, bins=xbins, **style)
+
+# plot the xdata locations on the x axis:
+ax.plot(xdata, 0*xdata, 'd')
+ax.set_ylabel('Number per bin')
+ax.set_xlabel('x bins (dx=1.0)')
+
+# %%
+# Modifying bins
+# ==============
+#
+# Changing the bin size changes the shape of this sparse histogram, so its a
+# good idea to choose bins with some care with respect to your data.  Here we
+# make the bins half as wide.
+
+xbins = np.arange(1, 4.5, 0.5)
+
+fig, ax = plt.subplots()
+ax.hist(xdata, bins=xbins, **style)
+ax.plot(xdata, 0*xdata, 'd')
+ax.set_ylabel('Number per bin')
+ax.set_xlabel('x bins (dx=0.5)')
+
+# %%
+# We can also let numpy (via Matplotlib) choose the bins automatically, or
+# specify a number of bins to choose automatically:
+
+fig, ax = plt.subplot_mosaic([['auto', 'n4']],
+                             sharex=True, sharey=True, layout='constrained')
+
+ax['auto'].hist(xdata, **style)
+ax['auto'].plot(xdata, 0*xdata, 'd')
+ax['auto'].set_ylabel('Number per bin')
+ax['auto'].set_xlabel('x bins (auto)')
+
+ax['n4'].hist(xdata, bins=4, **style)
+ax['n4'].plot(xdata, 0*xdata, 'd')
+ax['n4'].set_xlabel('x bins ("bins=4")')
+
+# %%
+# Normalizing histograms: density and weight
+# ==========================================
+#
+# Counts-per-bin is the default length of each bar in the histogram.  However,
+# we can also normalize the bar lengths as a probability density function using
+# the ``density`` parameter:
+
+fig, ax = plt.subplots()
+ax.hist(xdata, bins=xbins, density=True, **style)
+ax.set_ylabel('Probability density [$V^{-1}$])')
+ax.set_xlabel('x bins (dx=0.5 $V$)')
+
+# %%
+# This normalization can be a little hard to interpret when just exploring the
+# data. The value attached to each bar is divided by the total number of data
+# points _and_ the width of the bin, and thus the values _integrate_ to one
+# when integrating across the full range of data.
+# e.g. (``density = counts / (sum(counts) * np.diff(bins))``),
+# and (``np.sum(density * np.diff(bins)) == 1``).
```

Comment:
```suggestion
# e.g. ::
# 
#    density = counts / (sum(counts) * np.diff(bins))
#    np.sum(density * np.diff(bins)) == 1
```

The in-line code snippets are hard to read the way they got wrapped, do as a code block?